### ISSUE TITLE
Publisher error. Timed out while attempting to publish.

### ISSUE THREAD

**Issue by carlosgargz** [opentok/opentok-linux-sdk-samples#37]
**Describe the bug**

We are trying to create a video session using Opentok Linux SDK with a webcam that works good on Linux (Raspberry PI 64 bits) (audio and video), the video from the webcam is displaying successfully on the screen in an SDL window, unfortunately, the code shows an error when the program wants to Publish the Audio and Video.

The code error is the following. 
```
Publisher error. Error code: Timed out while attempting to publish.
Publisher error. Error string: 1541
```

**To Reproduce**
- Clone Basic_Video_Chat 
- Set the property session id, api key and token
- Plug in a webcam with an audio and video capability to a raspberry pi
- Compile the app and library corresponding to the 64 bits
- Launch basic_video_chat executable

**Expected behavior**
Windows created and stream published

**Screenshots**

The following is the window created and opened.

**Note:  Ignore the red close button icon, we modify the original renderer SDL 2 only to add an "exit" button to the user

![Window error](https://github.com/opentok/opentok-linux-sdk-samples/assets/91073175/f29c1047-1c1c-45e8-a5eb-e8c7404b9eb6)

**Device (please compete the following information):**
- sessionId: 2_MX40NzUxOTI1MX5-MTY4OTM2OTU4MTk2MX5VUEtkQ3QwSC9BZllZOWJ0cTVNMWZoRFp-fn4
- Linux SDK version: libopentok_linux_llvm_armv7-2.25.1
- OS and version: Linux raspberrypi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
- OS Name: Raspbian GNU/Linux 11 (bullseye)

**Additional context**

***** OpenTok Linux SDK | Version/Revision: linux_llvm-2.25.1/1a54d6ba20289af3387f37284ea9d22a17e92fd5 | Library built on: May 12 2023 | CA bundle version: Tue Jan 10 04:12:06 2023 GMT | Copyright 2023 Vonage | https://www.vonage.com/legal/communications-apis/terms-of-use/
on_session_connected callback function
on_publisher_error callback function
Publisher error. Error code: Timed out while attempting to publish.
Publisher error. Error string: 1541

Log: [error_log.txt](https://github.com/opentok/opentok-linux-sdk-samples/files/12054023/error_log.txt)

**Comment by v-kpheng** [comment]
@carlosgargz, apologies for the belated reply.

Looking at the logs, looks like you were able to connect to a session.  @juliobecerragomez, any ideas about the publishing error.

**Comment by juliobecerragomez** [comment]
@carlosgargz Does the publisher have an audio track? I assume it does. I remember reproducing this issue due to hardware issues. As it seems the camera is working for you, what about the audio system? Does it work with a video-only publisher?

**Comment by carlosgargz** [comment]
@v-kpheng Yes, that is correct I was able to get a session ID from the server and connect to the session, I think it is showing that message in the log file.

We have a webcam with microphone plugged in, when we execute the file the program seems to work because a new window pops up and show the video from the webcam, unfortunately, the program is not able to publish the audio (I can notice this because the C++ nor the logs reach the line "session published callback").

@juliobecerragomez Yes we have a webcam with microphone working in Raspbian OS, we tested the camera and the microphone with Audacity and VLC Media and worked.

We haven't tried the video-only publisher yet, we are going to test the video-only publisher.
